src/slider/slider-labels.tsx

```diff
@@ -0,0 +1,107 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import clsx from 'clsx';
+
+import styles from './styles.css.js';
+import { useMobile } from '../internal/hooks/use-mobile/index.js';
+import customCssProps from '../internal/generated/custom-css-properties/index.js';
+
+export interface SliderLabelProps {
+  min: number;
+  max: number;
+  referenceValues?: Array<number>;
+  valueFormatter?: (value: number) => string;
+  labelsId: string;
+}
+export default function SliderLabels({ min, max, referenceValues, valueFormatter, labelsId }: SliderLabelProps) {
+  const isMobile = useMobile();
+  const LABEL_THRESHOLD = isMobile ? 3 : 8;
+
+  const getVisibleReferenceValues = () => {
+    if (!referenceValues || referenceValues.length === 0) {
+      return [];
+    }
+
+    const values: number[] = [];
+    let lastValue = 0;
+    for (let i = 0; i <= referenceValues.length; i++) {
+      if (
+        Math.abs(referenceValues[i] - lastValue) >= (max - min) / LABEL_THRESHOLD &&
+        Math.abs(max - referenceValues[i]) >= (max - min) / LABEL_THRESHOLD
+      ) {
+        values.push(referenceValues[i]);
+        lastValue = referenceValues[i];
+      }
+    }
+    return values;
+  };
+
+  function getLabelPosition(index: number) {
+    const minDistance = (max - min) / LABEL_THRESHOLD;
+    const colSpan = Math.floor(minDistance / 2);
+    const positionStart = index - colSpan;
+    const positionEnd = index + colSpan;
+
+    return {
+      min: colSpan * 2,
+      max: (max - colSpan) * 2,
+      // add one to center the label
+      posStart: (positionStart - min) * 2 + 1,
+      posEnd: (positionEnd - min) * 2,
+    };
+  }
+
+  return (
+    <div
+      role="list"
+      className={clsx(styles.labels, {
+        [styles['labels-noref']]: !referenceValues,
+      })}
+      style={{
+        [customCssProps.sliderLabelCount]: !referenceValues ? 2 : (max - min) * 2,
+      }}
+      id={labelsId}
+    >
+      <span
+        role="listitem"
+        className={clsx(styles.label, styles['labels-min'])}
+        style={{
+          [customCssProps.sliderMinEnd]: max - min > LABEL_THRESHOLD ? getLabelPosition(0).min : 1,
+        }}
+      >
+        {valueFormatter ? valueFormatter(min) : min}
+      </span>
+      {getVisibleReferenceValues().map((step, index) => {
+        return (
+          <span
+            role="listitem"
+            key={`step-${index}`}
```

Comment:
nit: the key can be just index, there is no need to add a suffix or a prefix to it unless you need to render multiple elements with the same index

Reply:
Hmm, I don't like using indices as keys. From the [React docs](https://react.dev/learn/rendering-lists#why-does-react-need-keys): 

> You might be tempted to use an item’s index in the array as its key. In fact, that’s what React will use if you don’t specify a key at all. But the order in which you render items will change over time if an item is inserted, deleted, or if the array gets reordered. Index as a key often leads to subtle and confusing bugs.

Since this is a read only array, I don't think this is possible, but still seems like good practice to not use them. I updated to just use `step`.